src/components/signal-viewer/renderer.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import * as vega from 'vega';
+
+import { View } from '../../constants';
+import Table from '../table';
+
+import './index.css';
+
+const KEY = '_signals';
+const header = ['Signal', 'Value'];
+
+interface Props {
+  view?: View;
+  debugPane: boolean;
+}
+
+export default class SignalViewer extends React.Component<Props> {
+  constructor(props) {
+    super(props);
+  }
+  public componentDidMount() {
+    Object.keys(this.props.view[KEY]).map(signal => {
+      this.props.view.addSignalListener(signal, () => {
+        this.forceUpdate();
```

Comment:
Hmm, instead of forcing an update of the whole component, we should instead just take the new value (which is being passed to the callback) and update only that one value.

Reply:
@domoritz The whole component here means only a table. We can't just update one table row out of it. Because changing any row value will call render again anyway.

Reply:
Yes, but react can reuse dom elements easier and we don't have to grab signal values from Vega when they don't change. It seems cleaner to me to only update what has actually changed. 

Another option would be to make a react component that takes the view and the name of a signal and then renders only a single row. What do you think?

Reply:
Yes, that'll be more efficient actually. Good idea! I'll try to implement it.